src/zjs_event.c

```diff
@@ -198,54 +216,50 @@ static jerry_value_t remove_all_listeners(const jerry_value_t function_obj,
                                           const jerry_value_t argv[],
                                           const jerry_length_t argc)
 {
-    event_t *ev;
-
-    jerry_value_t event_emitter = zjs_get_property(this, HIDDEN_PROP("event"));
-    if (!jerry_get_object_native_handle(event_emitter, (uintptr_t*)&ev)) {
-        jerry_release_value(event_emitter);
-        ERR_PRINT("native handle not found\n");
-        return jerry_acquire_value(this);
-    }
-    jerry_release_value(event_emitter);
-
     if (!jerry_value_is_string(argv[0])) {
-        ERR_PRINT("event name must be first parameter\n");
-        return jerry_acquire_value(this);
+        return zjs_error("event name must be first parameter");
     }
 
+    jerry_value_t event_emitter = zjs_get_property(this, HIDDEN_PROP("event"));
+
     jerry_size_t size = ZJS_MAX_EVENT_NAME_SIZE;
     char event[size];
     zjs_copy_jstring(argv[0], event, &size);
     if (!size) {
-        ERR_PRINT("event name is too long\n");
-        return jerry_acquire_value(this);
+        jerry_release_value(event_emitter);
+        return zjs_error("event name is too long");
     }
 
+    jerry_value_t map = zjs_get_property(event_emitter, "map");
+    jerry_value_t event_obj = zjs_get_property(map, event);
+
     // Event object to hold callback ID and eventually listener arguments
-    jerry_value_t event_obj = zjs_get_property(ev->map, event);
     if (!jerry_value_is_object(event_obj)) {
+        jerry_release_value(event_emitter);
         jerry_release_value(event_obj);
-        DBG_PRINT("event object not found for '%s'\n", event);
-        return jerry_acquire_value(this);
+        jerry_release_value(map);
+        ERR_PRINT("event object not found\n");
+        return ZJS_UNDEFINED;
     }
 
-    int32_t callback_id = -1;
-    jerry_value_t id_prop = zjs_get_property(event_obj, "callback_id");
-    if (jerry_value_is_number(id_prop)) {
-        // If there already is an event object, get the callback ID
-        zjs_obj_get_int32(event_obj, "callback_id", &callback_id);
+    int32_t callback_id = get_callback_id(event_obj);
 
+    if (callback_id != -1) {
         zjs_remove_callback(callback_id);
 
         jerry_value_t name = jerry_create_string((const jerry_char_t*)event);
-        jerry_delete_property(ev->map, (const jerry_value_t)name);
+        jerry_delete_property(map, (const jerry_value_t)name);
         jerry_release_value(name);
     } else {
         ERR_PRINT("callback_id not found for '%s'\n", event);
     }
 
-    jerry_release_value(id_prop);
+    zjs_obj_add_number(event_emitter, 0, "numEvents");
```

Comment:
numEvents is different than the number of listeners. Events doesnt actually store the number of listeners, that is stored in a callback list, so when getNumListeners() is called we just ask the callback list how many functions there are. 

numEvents is the number of events registered to a listener, like onread, onerror, ondata etc.